File: ADS_2_POO/ADS2/InscricaoEstadual.cs

public class InscricaoEstadual : Documentos
{

    public InscricaoEstadual(string numeroInput) : base(numeroInput, 10, 8)
    {
        pesoUm = new int[] { 3, 2, 7, 6, 5, 4, 3, 2 };
        pesoFinal = new int[] { 4, 3, 2, 7, 6, 5, 4, 3, 2 };
        numero = DocumentosFormatacao.FormataNumero(numero, tamanhoComDigitos, tamanhoSemDigitos);
        numeroCorreto = CalcularNumeroCorreto(numero);
        mascaraNumero = numeroCorreto.Substring(0, 3) + "." + numeroCorreto.Substring(3, 5) + "-" + numeroCorreto.Substring(8, 2);
    }

 
}

File: ADS_2_POO/Python Version/inscricaoEstadual.py
from documentos import Documentos
from documentosFormatacao import DocumentosFormatacao

class InscricaoEstadual(Documentos):
    def __init__(self, numero_input):
        super().__init__(numero_input, 10, 8)
        self.peso_um = [3, 2, 7, 6, 5, 4, 3, 2]
        self.peso_final = [4, 3, 2, 7, 6, 5, 4, 3, 2]
        self.numero = DocumentosFormatacao.formata_numero(numero_input, self.tamanho_com_digitos, self.tamanho_sem_digitos)
        self.numero_correto = self.calcular_numero_correto(self.numero)
        self.mascara_numero = f"{self.numero_correto[:3]}.{self.numero_correto[3:8]}-{self.numero_correto[8:10]}"
